src/coreclr/jit/lower.cpp

```diff
@@ -4516,7 +4516,11 @@ GenTree* Lowering::LowerNonvirtPinvokeCall(GenTreeCall* call)
         switch (lookup.accessType)
         {
             case IAT_VALUE:
-                if (!IsCallTargetInRange(addr))
+                // IsCallTargetInRange always return true on x64. It wants to use rip-based addressing
+                // for this call. Unfortunately, in case of pinvokes (+suppressgctransition) to external libs
+                // (e.g. kernel32.dll) the relative offset is unlikely to fit into int32 and we will have to
+                // turn fAllowRel32 off globally.
+                if (call->IsSuppressGCTransition() || !IsCallTargetInRange(addr))
```

Comment:
This whole problem only exist with the JIT. We may want to skip the extra condition for `CORJIT_FLAG_PREJIT`. It is unnecessary deoptimization there.